posthog-android/src/main/java/com/posthog/android/PostHogAndroid.kt

```diff
@@ -79,6 +81,10 @@ public class PostHogAndroid private constructor() {
             config.replayStoragePrefix = config.replayStoragePrefix ?: replayPath.absolutePath
             val preferences = config.cachePreferences ?: PostHogSharedPreferences(context, config)
             config.cachePreferences = preferences
+            // Defaults to PostHogDeviceDateProvider when api < 33
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                config.dateProvider = (config.dateProvider as? PostHogAndroidDateProvider) ?: PostHogAndroidDateProvider()
```

Comment:
```suggestion
                config.dateProvider = if (config.dateProvider is PostHogDeviceDateProvider) PostHogAndroidDateProvider() else config.dateProvider
```

I think like this makes it more readable?

Reply:
the problem with this is that `PostDeviceDateProvider` is internal to the `posthog` module

Reply:
we can make it public and mark it as internal, like a few other ones, use the `@PostHogInternal` annotation